envoy/common/union_string.h

```diff
@@ -63,7 +63,7 @@ template <class Validator> class UnionStringBase {
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wuninitialized"
 #endif
-  UnionStringBase() : buffer_(InlinedStringVector()) {
+  UnionStringBase() : buffer_(std::in_place_type<InlinedStringVector>) {
```

Comment:
nice!
is the GCC ignore macros above (and enablement after) are still needed?
Would be great to clean these up!

Reply:
Yeah, I think we can clean it up.